presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeWindows.java

```diff
@@ -124,5 +139,15 @@ private static WindowNode collapseWindows(PlanNode source, WindowNode.Specificat
                     canonical.getPrePartitionedInputs(),
                     canonical.getPreSortedOrderPrefix());
         }
+
+        private static boolean dependsOn(WindowNode parent, WindowNode dependent)
+        {
+            return parent.getPartitionBy().stream().anyMatch(symbol -> dependent.getCreatedSymbols().contains(symbol))
```

Comment:
It would be nice to use `DependencyExtractor` here.

In regards of directed graph explained in the other comment. Then an edge could contain a list of dependant symbols which are different between nodes.